lib/lndclient/LndClient.ts

```diff
@@ -1004,30 +1004,46 @@ class LndClient extends SwapClient {
   /**
    * Attempts to close an open channel.
    */
-  public closeChannel = (fundingTxId: string, outputIndex: number, force: boolean): void => {
-    if (!this.lightning) {
-      throw(errors.UNAVAILABLE(this.currency, this.status));
-    }
-    const request = new lndrpc.CloseChannelRequest();
-    const channelPoint = new lndrpc.ChannelPoint();
-    channelPoint.setFundingTxidStr(fundingTxId);
-    channelPoint.setOutputIndex(outputIndex);
-    request.setChannelPoint(channelPoint);
-    request.setForce(force);
-    this.lightning.closeChannel(request, this.meta)
-      // TODO: handle close channel events
-      .on('data', (message: string) => {
-        this.logger.info(`closeChannel update: ${message}`);
-      })
-      .on('end', () => {
-        this.logger.info('closeChannel ended');
-      })
-      .on('status', (status: string) => {
-        this.logger.debug(`closeChannel status: ${JSON.stringify(status)}`);
-      })
-      .on('error', (error: any) => {
-        this.logger.error(`closeChannel error: ${error}`);
-      });
+  public closeChannel = (fundingTxId: string, outputIndex: number, force: boolean): Promise<void> => {
+    return new Promise<void>((resolve, reject) => {
+      if (!this.lightning) {
+        throw(errors.UNAVAILABLE(this.currency, this.status));
+      }
+      const request = new lndrpc.CloseChannelRequest();
+      const channelPoint = new lndrpc.ChannelPoint();
+      channelPoint.setFundingTxidStr(fundingTxId);
+      channelPoint.setOutputIndex(outputIndex);
+      request.setChannelPoint(channelPoint);
+      request.setForce(force);
+
+      this.lightning.closeChannel(request, this.meta)
+        .on('data', (message: lndrpc.CloseStatusUpdate) => {
+          if (message.hasClosePending()) {
+            const txId = message.getClosePending()?.getTxid_asB64();
+            if (txId) {
+              this.logger.info(`channel closed with tx id ${txId}`);
```

Comment:
Have you tested this code? Does it output the transaction id in the usual hex format? I have had some mixed experiences with getting transactions ids from the LND gRPC

Reply:
No this would actually print in base64, which is not ideal I realize as far as looking up a transaction based on what's in the logs. I'll make this use hex and also give it a quick test to ensure the output is good.

Reply:
I confirmed that this is outputting hex now.